nservicebus/upgrades/nhibernate-timeouts-index.md

```diff
@@ -0,0 +1,124 @@
+---
+title: NHibernate Persistence - Resolving incorrect timeout table indexes #252
+summary: Instructions on how to resolve incorrect index definitions that can cause performance degradation for affected versions 6 to 7.
+component: NHibernate
+isUpgradeGuide: true
+upgradeGuideCoreVersions:
+ - 5
+ - 6
+---
+
+
+## Summary
+
+This guidance explains how to resolve an incorrectly created index when passing a custom NHibernate configuration to NServiceBus as described in the following issue:
+
+- https://github.com/Particular/NServiceBus.NHibernate/issues/252
+
+This issue causes performance degradation if the table contains a large number of rows. Inserts and queries are inefficient due to the incorrect order of columns. This results in unnecessary locking which limits the processing throughput of timeouts.
+
+
+## Compatibility
+
+This issue has been resolved in the following patch versions as defined in the NServiceBus [support policy](support-policy.md):
+
+- NServiceBus.NHibernate 7.1.4
+- NServiceBus.NHibernate 7.0.6
+- NServiceBus.NHibernate 6.2.8
+
+
+If any of the supported affected minor versions (7.1.x, 7.0.x, or 6.2.8) are used these should be updated to the latest patch release. If an older version - non supported - affected version is used, this should be updated to a newer minor (in case of 6.1.x or 6.0.x) or major version (any version prior to 6.x).
+
+
+## Upgrade steps
+
+Steps:
+
+ * Update to latest patch release
+ * Deploy the new version
+ * Check if a warning related to this schema issue is visible
+   * Or manually inspect the schema in the database
+ * Follow the procedure on how to resolve schema issues for the database engine used (Microsoft SQL Server or Oracle)
+   * If any other database engine is used then these changes must be applied manually
+
+
+## Check at startup
+
+If there are endpoints that created an incorrect index definition then this is detected in all fixed supported versions for 6.2.x, 7.0.x and 7.1.x. The detection routine is run when the endpoint instance is created and started. If that instance is affected the following log event with log level warning is written:
```

Comment:
> If that instance is affected the following log event with log level warning is written:

This could be simplified:

> If that instance is affected the following warning is logged: